test/qtismtest/data/results/ContextTest.php

```diff
@@ -33,21 +34,38 @@ public function testAddSessionIdentifier()
         $sourceId = 'a source id';
         $identifier = "string with\n\rtabulations\tand\r\nnew lines\nto be replaced\rby spaces";
         $normalizedIdentifier = 'string with  tabulations and  new lines to be replaced by spaces';
-        
+
         $subject = new Context();
         $this->assertFalse($subject->hasSessionIdentifiers());
-        
+
         $subject->addSessionIdentifier($sourceId, $identifier);
         $this->assertTrue($subject->hasSessionIdentifiers());
-        
+
         $sessionIdentifierCollection = $subject->getSessionIdentifiers();
         $this->assertCount(1, $sessionIdentifierCollection);
-        
+
         $sessionIdentifier = $sessionIdentifierCollection->current();
         $this->assertInstanceOf(SessionIdentifier::class, $sessionIdentifier);
         /** @var SessionIdentifier $sessionIdentifier */
-        
+
         $this->assertEquals($sourceId, $sessionIdentifier->getSourceID()->getValue());
         $this->assertEquals($normalizedIdentifier, $sessionIdentifier->getIdentifier()->getValue());
     }
+
+    public function testAddSessionIdentifierWithDuplicateSourceIdThrowsException()
+    {
+        $sourceId = 'a source id';
+        $identifier1 = 'id1';
+        $identifier2 = 'id2';
+
+        $subject = new Context();
+        $this->assertFalse($subject->hasSessionIdentifiers());
```

Comment:
please add a unit test with non identical source id, but identical identifier (should pass)